nestkernel/event.h

```diff
@@ -996,6 +1000,21 @@ class DataSecondaryEvent : public SecondaryEvent
     supported_syn_ids_.push_back( synid );
   }
 
+  const std::vector< synindex >&
+  get_supported_syn_ids() const
+  {
+    return supported_syn_ids_;
+  }
+
+  void clear_supported_syn_ids()
```

Comment:
At some point in this file there should probably be some comment explaining why there is an additional vector `pristine_supported_syn_ids_` and why this is needed. From what I get from a short investigation this is to ensure that additional syn_ids that result from e.g. CopyModel are removed when ResetKernel is called, right? If I am right here this basicly fixes a (so far undetected) existing bug in the handling of SecondaryEvents, nice!

Reply:
yes, this is the idea. we need to reset the `supported_syn_ids_` vector, otherwise we can run into issues for sequences of calls like `ResetKernel ... CopyModel ... Simulate ... ResetKernel ... Simulate`. will add documentation for this